download-basket/src/main/java/downloadbasket/helpers/DownloadServices.java

```diff
@@ -0,0 +1,213 @@
+package downloadbasket.helpers;
+
+import fi.nls.oskari.log.LogFactory;
+
+import fi.nls.oskari.log.Logger;
+import fi.nls.oskari.util.PropertyUtil;
+import fi.nls.oskari.util.IOHelper;
+import downloadbasket.data.ErrorReportDetails;
+import downloadbasket.data.LoadZipDetails;
+import org.apache.commons.mail.MultiPartEmail;
+
+import javax.activation.DataHandler;
+import javax.activation.DataSource;
+import javax.activation.FileDataSource;
+import javax.mail.internet.MimeBodyPart;
+import javax.mail.internet.MimeMultipart;
+import javax.mail.util.ByteArrayDataSource;
+import java.io.*;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.UUID;
+import java.util.Locale;
+import java.util.zip.ZipException;
+import java.util.zip.ZipFile;
+import org.springframework.context.MessageSource;
+import fi.nls.oskari.spring.SpringContextHolder;
+
+/**
+ * Download services.
+ *
+ */
+public class DownloadServices {
+	public static final String WFS_GETCAPABILITIES_XML_JSON_VALUE = "WFSGetCapabilitiesXML";
+	public static final String WFS_METADATA_URL_JSON_VALUE = "linkForMetadata";
+	public static final String WFS_USED_URL = "url";
+	public static final String WFS_FEATURETYPES = "featureTypes";
+	private final Logger LOGGER = LogFactory.getLogger(DownloadServices.class);
+
+	private MessageSource messages;
+
+	/**
+	 * Default Constructor.
+	 */
+	public DownloadServices() {
+	}
+
+	/**
+	 * Load shape-ZIP from Geoserver.
+	 * 
+	 * @param ldz
+	 *            load zip details
+	 * @return filename file name
+	 * @throws IOException
+	 * 
+	 *             Normal way download uses BBOX as the cropping method.
+	 *             Otherwise, filter plugin is used.
+	 * 
+	 */
+	public String loadZip(LoadZipDetails ldz, Locale locale) throws IOException {
+		String realFileName = "";
+		String returnFileName = "";
+		HttpURLConnection conn = null;
+
+		try {
+			LOGGER.debug("WFS URL: " + ldz.getWFSUrl());
+
+			if (ldz.isDownloadNormalWay()) {
+				LOGGER.debug("Download normal way");
+			} else {
+				LOGGER.debug("Download plugin way");
```

Comment:
This else block is never hit as downloadNormalWay is always true in LoadZipDetails due to the setter not actually setting anything.